### DISCUSSION TITLE
Fullstack .NET

### DISCUSSION THREAD

**Discussion by rogercomp** [balta-io/forum#679] [Tira Dúvidas] [answered]
Ao publicar a aplicação em um servidor nginx ou Azure ou no Github e ao clicar em sair da aplicação está estourando um erro na aplicação
"_Constructor Contains Null Parameter Names_"
no metodo(OnInitializedAsync) da pagina Logout:  
 protected override async Task OnInitializedAsync()
 {
     try
     {
         if (await AuthenticationStateProvider.CheckAuthenticatedAsync())
         {
             **await Handler.LogoutAsync();**
             await AuthenticationStateProvider.GetAuthenticationStateAsync();                
             AuthenticationStateProvider.NotifyAuthenticationStateChanged();
         }
     }
     catch (Exception ex)
     {
         Snackbar.Add(ex.Message, Severity.Error);
     }    
 }

Esse erro so acontece ao publicar a aplicação, localmente o erro não acontece.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @rogercomp, como vai?

Quais passos está executando? Consegue enviar um print ou mais detalhes do erro?

**Reply by rogercomp** [reply]
1 - Gero o publish para linux com dotnet publish -c Release -r linux-x64
2 - coloco na pasta de publicação configurada no nginx
3 - Acesso o sistema normalmente, navego nos menus 
4 - Ao clicar em sair da mensagem "Constructor Contains Null Parameter Names" e não consegue limpar o cookie

**Reply by rogercomp** [reply]
![erro](https://github.com/user-attachments/assets/cd79ea23-7733-4664-917c-de1515bf68e2)

**Reply by andrebaltieri** [reply]
No console não aparece mais detalhes?

**Reply by rogercomp** [reply]
Localmente não aparece nada

**Reply by andrebaltieri** [reply]
Mas no console em produção... F12 ai no Browser!

**Reply by rogercomp** [reply]
![image](https://github.com/user-attachments/assets/39e7e812-e8bb-405e-ae0a-1c221d48f17a)


Sem mensagens de requsições com erro

**Reply by andrebaltieri** [reply]
Estranho, pesquisei e parece um erro de serialização... Tem certeza que seu App (Frontend) tem acesso a API (Backend)? Você consegue testar alguma requisição via Postman por exemplo para garantir que a API está funcionando?

Em adicional, localmente, consegue derrubar a API e rodar seu App para tentar causar um erro parecido?

**Comment by rogercomp** [comment]
Sim

Em seg., 11 de nov. de 2024, 10:11, Andre Baltieri ***@***.***>
escreveu:

**Reply by andrebaltieri** [reply]
Sim o que?
